src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,40 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final String SPLITTER = " ";
+    private static final String HYPHEN = " - ";
+    private final Map<String, Integer> employees = new HashMap<>();
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        for (String name : names) {
+            employees.put(name, 0);
+        }
+        StringBuilder reportForPeriod = new StringBuilder("Report for period ").append(dateFrom)
+                .append(HYPHEN).append(dateTo);
+        for (String employeeData : data) {
+            String[] details = employeeData.split(SPLITTER);
+            LocalDate date = getDate(details[0]);
+            String nameOfEmployee = details[1];
+            int dailySalary;
+            if (date.compareTo(getDate(dateFrom)) >= 0 && date.compareTo(getDate(dateTo)) <= 0) {
```

Comment:
getDate(dateTo) and getDate(dateFrom) can be called only once in the very beginning